src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Converters.cs

```diff
@@ -48,9 +48,9 @@ public JsonConverter GetConverter(Type typeToConvert)
 
             if (_typeInfoResolver is null)
             {
-                // Backward compatibility -- root the default reflection converters
+                // Backward compatibility -- root & query the default reflection converters
                 // but do not populate the TypeInfoResolver setting.
-                DefaultJsonTypeInfoResolver.RootDefaultInstance();
+                return DefaultJsonTypeInfoResolver.GetConverterForType(typeToConvert, this);
```

Comment:
will this throw for builtin converters if we didn't root?

Reply:
No. That method will root the built-in converters itself since it's already marked RUC.